homeassistant/components/weatherflow/__init__.py

```diff
@@ -0,0 +1,77 @@
+"""The WeatherFlow integration."""
+from __future__ import annotations
+
+from pyweatherflowudp.client import EVENT_DEVICE_DISCOVERED, WeatherFlowListener
+from pyweatherflowudp.device import EVENT_LOAD_COMPLETE, WeatherFlowDevice
+from pyweatherflowudp.errors import ListenerError
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    EVENT_HOMEASSISTANT_STARTED,
+    EVENT_HOMEASSISTANT_STOP,
+    Platform,
+)
+from homeassistant.core import CoreState, Event, HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import DOMAIN, LOGGER
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up WeatherFlow from a config entry."""
+    client = WeatherFlowListener(host=entry.data.get(CONF_HOST, "0.0.0.0"))
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = client
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    @callback
+    def device_discovered(device: WeatherFlowDevice) -> None:
+        LOGGER.debug("Found a device: %s", device)
+
+        @callback
+        def add_device() -> None:
+            async_dispatcher_send(
+                hass, f"{DOMAIN}_{entry.entry_id}_add_{SENSOR_DOMAIN}", device
+            )
+
+        entry.async_on_unload(
+            device.on(
+                EVENT_LOAD_COMPLETE,
+                lambda _: add_device()
+                if hass.state == CoreState.running
+                else hass.bus.async_listen_once(
+                    EVENT_HOMEASSISTANT_STARTED, lambda _: add_device()
+                ),
```

Comment:
Please break out the ternary into an if block above

Reply:
You can use `from homeassistant.helpers.start import async_at_start`

Reply:
Does it just become this:

```python
        entry.async_on_unload(
            device.on(EVENT_LOAD_COMPLETE, async_at_start(hass, add_device()))
        )
```

Reply:
Much cleaner!